src/workers.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileoverview A Worker pool comprising of WebWorkers.
+ * https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers
+ */
+
+WebWorkerPool = function(options) {
+  this.workers_ = [];
+  this.free_workers_ = [];
+  this.work_queue_ = [];
+
+  initWorkers(options);
+};
+
+WebWorkerPool.prototype.postMessage = function(fn, on_done) {
+  if (this.free_workers_.length > 0) {
+    worker = this.free_workers_.pop();
+    worker.port.postMessage(new WorkUnit(fn, on_done, worker));
+  } else {
+    work_queue_.add(new WorkUnit(fn, on_done, null));
+  }
+};
+
+WebWorkerPool.prototype.initWorkers = function(options) {
+  for (var i = 0; i < options.worker_pool_size; ++i) {
+    worker = new SharedWorker("worker.js"));
```

Comment:
From the docs, it seems like the way to go. We would want the worker threadPool to be accessible by all the scripts on the page.